src/common/local_file_system.cpp

```diff
@@ -934,7 +935,9 @@ vector<string> LocalFileSystem::Glob(const string &path, FileOpener *opener) {
 		absolute_path = true;
 	} else if (splits[0] == "~") {
 		// starts with home directory
-		auto home_directory = GetHomeDirectory(opener);
+		auto info = duckdb::make_uniq<FileOpenerInfo>();
```

Comment:
You can probably just make this on the stack instead of the heap i.e.

```cpp
FileOpenerInfo info = {path};
auto home_directory = GetHomeDirectory(opener, &info);
```